File: appfw/src/main/java/com/jx/appfw/data/net/BaseApiImpl.java
package com.jx.appfw.data.net;

import com.jx.arch.net.ApiConnection;

import java.net.MalformedURLException;

/**
 * 网络请求处理类封装
 */
public class BaseApiImpl
{
    /**
     * POST请求
     *
     * @param url
     * @param requestJson
     * @return
     * @throws MalformedURLException
     */
    public String requestFromApi(String url, String requestJson) throws MalformedURLException
    {
        return ApiConnection.createPost(url, requestJson).requestPostSyncCall();
    }

    /**
     * GET请求
     *
     * @param url
     * @return
     * @throws MalformedURLException
     */
    public String requestFromApi(String url) throws MalformedURLException
    {
        return ApiConnection.createGET(url).requestGetSyncCall();
    }
}

File: appfw/src/main/java/com/jx/appfw/data/net/BaseApiImpl.kt
package com.jx.appfw.data.net

import com.jx.appfw.net.ApiConnection
import java.net.MalformedURLException

open class BaseApiImpl {
    /**
     * POST请求
     *
     * @param url
     * @param requestJson
     * @return
     * @throws MalformedURLException
     */
    @Throws(MalformedURLException::class)
    fun requestFromApi(url: String, requestJson: String): String? {
        return ApiConnection.createPost(url, requestJson)
    }

    /**
     * GET请求
     *
     * @param url
     * @return
     * @throws MalformedURLException
     */
    @Throws(MalformedURLException::class)
    fun requestFromApi(url: String): String? {
        return ApiConnection.createGet(url)
    }
}